frontend/benefit/applicant/src/pages/index.tsx

```diff
@@ -31,6 +34,14 @@ const ApplicantIndex: NextPage = () => {
     };
   }, [setIsNavigationVisible]);
 
+  /**
+   * Fix a cache issue where single application's button opens up a read-only
+   * application if status changes to "additional_information_needed"
+   */
+  const effectInvalidateApplicationsCache = () =>
+    void queryClient.invalidateQueries('applications');
+  useEffect(effectInvalidateApplicationsCache, [queryClient]);
```

Comment:
The exact conditions that make the query client change and thus trigger this are a bit unclear to me, but if this works, sure.

Reply:
For some reason the navigation results to react-query fetching data from cache even though all requests should be considered as non-stale:

![image](https://github.com/City-of-Helsinki/yjdh/assets/5328394/4c078359-a43c-4fea-80be-0bd367a1ea19)

I consider it safe to just clear any leftover react-query cache here as it's never meant to work with cache.

https://github.com/TanStack/query/discussions/4666